src/inference/hmc.jl

```diff
@@ -33,51 +75,93 @@ sample(gdemo([1.5, 2]), HMC(1000, 0.01, 10))
 ```
 """
 mutable struct HMC{AD, space, metricT <: AHMC.AbstractMetric} <: StaticHamiltonian{AD}
-    n_iters     ::  Int       # number of samples
     ϵ           ::  Float64   # leapfrog step size
     n_leapfrog  ::  Int       # leapfrog step number
 end
+
+transition_type(::Sampler{<:Hamiltonian}) = Transition
+alg_str(::Sampler{<:Hamiltonian}) = "HMC"
+
 HMC(args...) = HMC{ADBackend()}(args...)
-function HMC{AD}(n_iters::Int, ϵ::Float64, n_leapfrog::Int, ::Type{metricT}, space::Tuple) where {AD, metricT <: AHMC.AbstractMetric}
-    return HMC{AD, space, metricT}(n_iters, ϵ, n_leapfrog)
+function HMC{AD}( ϵ::Float64, n_leapfrog::Int, ::Type{metricT}, space::Tuple) where {AD, metricT <: AHMC.AbstractMetric}
+    return HMC{AD, space, metricT}(ϵ, n_leapfrog)
 end
 function HMC{AD}(
-    n_iters::Int,
     ϵ::Float64,
     n_leapfrog::Int,
     ::Tuple{};
     kwargs...
 ) where AD
-    return HMC{AD}(n_iters, ϵ, n_leapfrog; kwargs...)
+    return HMC{AD}(ϵ, n_leapfrog; kwargs...)
 end
 function HMC{AD}(
-    n_iters::Int,
     ϵ::Float64,
     n_leapfrog::Int,
     space::Symbol...;
     metricT=AHMC.UnitEuclideanMetric
 ) where AD
-    return HMC{AD}(n_iters, ϵ, n_leapfrog, metricT, space)
+    return HMC{AD}(ϵ, n_leapfrog, metricT, space)
+end
+
+function sample_init!(
+    rng::AbstractRNG,
+    model::Model,
+    spl::Sampler{<:Hamiltonian},
+    N::Integer;
+    adaptor=AHMCAdaptor(spl.alg),
```

Comment:
Do we every use this default value of keyword arugment?

Reply:
No, I don't think so. Perhaps I should move this to where the `HMCState` is constructed, so that users can pass in a separate adaptor if they want?

Reply:
Yes. Let's remove it first and maybe improve the way set adaptor later.